src/dash-table/utils/TableClipboardHelper.ts

```diff
@@ -21,6 +23,7 @@ export default class TableClipboardHelper {
         );
 
         const value = SheetClip.prototype.stringify(df);
+        TableClipboardHelper.lastLocalCopy = df;
```

Comment:
Turns out that SheetClip saves undefined|null as a empty string. Which is probably fine when copying to/from Excel but causing unexpected behavior when copying from a table to another (in the same context).

Here, keeping a copy of the table's df slice used for the copy and comparing its stringify result to the value in the clipboard on paste. If the values are identical, making the assumption that the copy/paste was initiated with table data.. reusing the df slice instead of the parse result, keeping the undefined|null values..

It's a bit hacky..